cigarettes cigars lighters fit in helmet cosmetic

hislittlecuzingames opened the pull request:
# About the pull request

Allows smoking rp items to fit in helmet cosmetic slots instead of taking up "valuable" slots.

# Explain why it's good for the game

If you fill up your slots with gear, you often cannot fit smoking items like cigarettes and lighters. This allows players to RP more with their ability to take RP items without sacrificing gameplay items.

**please verify...**
I thought you'd use pipe equals to add bitflags in dm ie
my-bitflag |= MYNEWVALUE
This wouldn't compile though...

I saw ampersand equals removes items not on the right side of equals.
So I used regular equals and it seems to fit into backpacks and webbing no problem.


# Testing Photographs and Procedure

I tested by spawning using "create humans" and spawning USCM and their class.
I verified the items fit in each of the basic helmets and also put in "game items" like bandages and an injector to assure they took the cosmetic slots, not the "item" slots.

I also put the cigarettes into the 5 slot webbing and backpacks.

<details>
<summary>Screenshots & Videos</summary>

![354072752-7921f9a0-51af-49e1-a521-a3f9f8cd6363](https://github.com/user-attachments/assets/a5086670-0adb-49dc-b903-fa4180fe1d2c)

![354072765-f37b936a-3763-4d84-b047-69116d8d9b18](https://github.com/user-attachments/assets/cc928f13-e100-4e6d-8dee-828d0e6928f8)

![354072809-c56198eb-7482-4814-803b-e52193446e94](https://github.com/user-attachments/assets/91116610-9589-4c58-9871-2efd21523516)

![354073089-adb46a73-3798-4441-b381-0ef4432df429](https://github.com/user-attachments/assets/2f20364d-569e-4eb1-9fe0-4ef629e4a252)

![354073414-2bfb7d21-feba-4d7a-8632-233add523e30](https://github.com/user-attachments/assets/ee69872a-feca-420a-b3ec-1ce2087d3bc1)


</details>


# Changelog
:cl:
qol: Can roleplay easier with cigarettes, cigars, lighters counting as cosmetic for helmet storage purposes.
/:cl:

Drulikar commented on code/game/objects/items/storage/fancy.dm:
```diff
@@ -269,6 +270,7 @@
 	throwforce = 2
 	w_class = SIZE_SMALL
 	flags_equip_slot = SLOT_WAIST
+	flags_obj = OBJ_IS_HELMET_GARB
```
An entire cigar case doesn't need to be a helmet accessory (though its too long to fit anyways). You likely intended `/obj/item/clothing/mask/cigarette` (and thus subtype `/obj/item/clothing/mask/cigarette/cigar`) instead.

hislittlecuzingames replied:
resolved.

Drulikar reviewed (changes requested):
> I thought you'd use pipe equals to add bitflags in dm ie
my-bitflag |= MYNEWVALUE
This wouldn't compile though...

The error you likely got was that you can't assign a non-constant value in a definition.

Doing `whatever |= thing` means take whatever already exists in `whatever` and bitwise or that with `thing`. Since the existing value can be anything, thats not a constant value; and thus cannot be in a definition.

Code that has to execute cannot be in a definition.

With 515 the `::` operator could do something similar though: e.g. `flags_obj = parent_type::flags_obj|OBJ_IS_HELMET_GARB`

But in this case the flags_object for all of these items, including `/obj/item/clothing/mask/cigarette` are already 0 so there is nothing to inherit for that flag currently.

hislittlecuzingames commented:
Thanks for the code snippet. 

I made the adjustment to the better code convention, removed cigar boxes and added matchboxes to be cosmetic slots in helmets. 
I verified they (cigarette pack, lighter) fit in the com tech helmet. 

re-requesting review.

Drulikar commented:
Pr title and description currently still mentions cigars but code does nothing regarding cigars.

hislittlecuzingames commented:
> Pr title and description currently still mentions cigars but code does nothing regarding cigars.

the cigars are under the cigarettes as you mentioned.
They're in the "flame tool" file for some reason. 
![image](https://github.com/user-attachments/assets/a816c84c-3266-49ec-8f1b-efed2a8757e6)

If you want me to test the cigar, I don't know where they are but if you tell me, I can produce screenshots of verified testing.

hislittlecuzingames commented:
> > Pr title and description currently still mentions cigars but code does nothing regarding cigars.
> 
> the cigars are under the cigarettes as you mentioned. They're in the "flame tool" file for some reason. ![image](https://private-user-images.githubusercontent.com/142365554/354647497-a816c84c-3266-49ec-8f1b-efed2a8757e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2MTMyNTYsIm5iZiI6MTcyMjYxMjk1NiwicGF0aCI6Ii8xNDIzNjU1NTQvMzU0NjQ3NDk3LWE4MTZjODRjLTMyNjYtNDllYy04ZjFiLWVmZWQyYTg3NTdlNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwMlQxNTM1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZjkzMjFiODUxMGUxODViZGQ1MzA0MWY3NDcxNDY0NmU3MGVlNTBhZGNjYjc4Yjg3MzNiNDEzZjg1NGRkNjRhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.X9o3osiSvT2gRJiO5P54xamLdJQxfjW8sQUNuI_W5RY)
> 
> If you want me to test the cigar, I don't know where they are but if you tell me, I can produce screenshots of verified testing.

nevermind I can't read. 

added appropriate support in "added cigar and weed" commit.
